### ISSUE TITLE
Refactor family provider listeners to be on the highest-hierarchy possible that has direct access to the family arguments

### ISSUE THREAD

**Issue by nazrinharris** [nazrinharris/pecunia#105]
The title is a mouthful, but basically, I want to remove the listeners from the top of the `DebugViewAccountScreen` and put it somewhere like `AccountsListTile`. Where if it is reused, it can still listen to providers with specific family arguments.

Okay, maybe this isn't a decent example, because I had `DeleteAccount` in mind, which is a class provider and isn't a family provider. I faced the issue where when I tried to edit a transaction, I couldn't show a dialog, because I couldn't listen on the specific `editTransaction` provider, because I didn't have access to the family arguments. By having the listener in the `TransactionListTile` itself, whenever something happens to it, I can easily add the listener to respond.

Rather than having the listeners in widgets like the list tile, it might be better to have them within widgets that actually does the writes. Like `EditTxnForm` and `CreateTxnForm`. Far more intuitive and simple. (Of course, in case of, delete transaction, for example, the listener would be in the `TxnBottomSheet`. But this may move up to `TxnListTile` if the user is able to delete `TxnListTile` right then and there)

To sum it up, a "to-do-list" of sorts is all the listens I'm doing in `DebugViewAccountScreen`. These should be done in the respective widgets. Like `CreateTransactionBottomSheet` for example. 

# Deed
### `CreateTxnBottomSheet` and `CreateTransferTxnBottomSheet`
- [x] Listen for when transaction is created
- [x] Listen for when transfer transaction is created

### Others
- [x] Listen for when transaction is deleted in `TxnBottomSheet`
- [x] Listen for when transaction is edited in `EditTxnForm`
- [x] Listen for when transfer transaction is deleted in `TransferTxnBottomSheet`
- [x] Listen for when transfer transaction is edited in `EditTransferTxnForm`

### `CreateAccountBottomSheet`
- [x] Listen for when account is created

**Comment by nazrinharris** [comment]
Do I put the listen in the `TxnBottomSheet` or should I put it in like `EditTxnFormWidget`?

**Comment by nazrinharris** [comment]
Having it in `EditTxnFormWidget` would make it very intuitive on where listens should be. I'll do that instead

**Comment by nazrinharris** [comment]
I also must clarify, there isn't a `CreateTxnBottomSheet` persay, because its only a`CreateTxnForm` and a corresponding `showCreateTxnBottomSheet` method

**Comment by nazrinharris** [comment]
Should be completed with 4be25f2 in #102. But this issue may be visited again by me in the future. So, hi from the old me!
